doc/source/security/data-at-rest-encryption.rst

```diff
@@ -0,0 +1,390 @@
+
+.. warning::
+
+    Each ``secret_mount_point`` must be used by only one server. Multiple
+    servers using the same secret_mount_point may cause unpredictable behavior.
+
+Create a backup of the keyring configuration file or data file immediately
+after creating the encrypted tablespace. If you are using Master key encryption, backup before master key rotation and after master key rotation.
+
+The first time a key is retrieved from a `keyring`, the `keyring_vault`
+communicates with the Vault server to retrieve the key type and data.
+
+System Variables
+--------------------
+
+.. variable:: keyring_vault_config
+
+    :cli: ``--keyring-vault-config``
+    :dyn: Yes
+    :scope: Global
+    :vartype: Text
+    :default:
+
+This variable is used to define the location of the :ref:`keyring_vault_plugin`
+configuration file.
+
+.. variable:: keyring_vault_timeout
+
+  :cli: ``--keyring-vault-timeout``
+  :dyn: Yes
+  :scope: Global
+  :vartype: Numeric
+  :default: ``15``
+
+Set the duration in seconds for the Vault server connection timeout. The
+default value is ``15``. The allowed range is from ``0`` to ``86400``. To wait an infinite amount of time set the variable to ``0``.
+
+Verifying the Keyring Plugin is Active
+---------------------------------------
+
+To verify the keyring plugin is active, run the `SHOW PLUGINS
+<https://dev.mysql.com/doc/refman/8.0/en/show-plugins.html>`__ statement or
+run a query on the `INFORMATION_SCHEMA.PLUGINS` table. You can also query the PLUGINS view.
+
+.. code-block:: mysql
+
+    mysql> SELECT plugin_name, plugin_status FROM INFORMATION_SCHEMA.PLUGINS WHERE plugin_name LIKE 'keyring%';
+
+    +---------------+----------------+
+    | plugin_name   | plugin_status  |
+    +===============+================+
+    | keyring_file  | ACTIVE         |
+    +---------------+----------------+
+
+Encrypting a File-Per-Table Tablespace
+--------------------------------------
+
+The `CREATE TABLESPACE <https://dev.mysql.com/doc/refman/5.7/en/create-tablespace.html>`_ statement is extended to allow the ``ENCRYPTION=['Y/N']`` option to encrypt a File-per-Table tablespace.
+
+.. code-block:: mysql
+
+    mysql> CREATE TABLE myexample (id INT mytext varchar(255)) ENCRYPTION='Y';
+
+To enable encryption to an existing tablespace, add the ``ENCRYPTION`` option to the ``ALTER TABLE`` statement.
+
+.. code-block:: mysql
+
+    mysql> CREATE TABLE myexample ENCRYPTION='Y';
+
+You must add the ``ENCRYPTION`` option to ``ALTER TABLE`` to change the table encryption state. Without the ``ENCRYPTION`` option, an encrypted table remains encrypted or an unencrypted table remains unencrypted.
+
+Encrypting a General Tablespace
+-------------------------------------------
+
+As of :rn:`5.7.20-18`, |Percona Server| supports general tablespace encryption. A general tablespace must be either encrypted; all tables are encrypted, or not encrypted, no encrypted tables. You cannot encrypt only some of the tables in a general tablespace.
```

Comment:
"no encrypted tables." - is this unessesary repetition, or you meat that - tables cannot be selectively encrypted when they are part of a tablespace ?

Reply:
Rewritten.